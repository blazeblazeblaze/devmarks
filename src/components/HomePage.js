import React from "react";
import FilteredTopics from "../containers/FilteredTopics";
import TopicFiltersForm from "../containers/TopicFiltersForm";
import Hero from "./Hero";

const HomePage = () => (
  <div>
    <Hero title="Browse Courses & Tutorials" subtitle="Type any phrase below" />
    <TopicFiltersForm />
    <FilteredTopics />
  </div>
);

export default HomePage;
