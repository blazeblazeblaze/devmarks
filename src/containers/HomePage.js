import React, { Component } from "react";
import { connect } from "react-redux";

import Home from "../components/Home";
import { filterTopics } from "../actions/filters";
import { getTopics } from "../actions/topics";

import { visibleTopics } from "../reducers";

const HomePage = class extends Component {
  componentDidMount() {
    this.props.requestTopics();
  }
  render() {
    return <Home {...this.props} />;
  }
};

export default connect(
  state => ({
    topics: visibleTopics(state.topics, state.topicsFilter.text),
    topicsFilter: state.topicsFilter
  }),
  {
    onSearch: searchTopic,
    requestTopics: getTopics
  }
)(HomePage);
