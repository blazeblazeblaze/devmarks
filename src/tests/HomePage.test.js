import React from "react";
import HomePage from "../components/HomePage";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<HomePage />", () => {
  const tree = shallow(<HomePage />);

  it("matches snapshot", () => {
    expect(toJson(tree)).toMatchSnapshot();
  });
});
