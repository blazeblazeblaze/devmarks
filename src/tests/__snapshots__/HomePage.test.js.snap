// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> matches snapshot 1`] = `
<div>
  <Hero
    subtitle="Type any phrase below"
    title="Browse Courses & Tutorials"
  />
  <Connect(TopicFilters) />
  <Connect(FilteredTopics) />
</div>
`;
